### DISCUSSION TITLE
`useObjectRef` may have an issue

### DISCUSSION THREAD

**Discussion by alirezamirian** [adobe/react-spectrum#3508] [Q&A]
`useObjectRef` syncs the forwarded ref with the returned object ref, in a layoutEffect, which has the forwarded ref as a dependency:
https://github.com/adobe/react-spectrum/blob/24afd2a1feeb81bb71001204e1f2c9fc986ae282/packages/%40react-aria/utils/src/useObjectRef.ts#L43

Since the ref object doesn't change across renders, it probably means the ref is synchronized **only after the first render**.  I haven't come up with a concrete problematic case, but it seems the layoutEffect, shouldn't have any dependency, letting the ref value be synchronized after each render.
I hesitated to create an issue, since I haven't checked the issue in an example yet.

**Comment by snowystinger** [comment]
if you find an issue, please feel free to open a PR to address it or even just add a test. we haven't found any problems with it as of yet, but that doesn't mean we can't improve it.
